### ISSUE TITLE
Latest commit breaks default font loading

### ISSUE THREAD

**Issue by Basez** [ocornut/imgui#109]
Hi, I just started using imgui and got weird font glitches. After a bit of debugging I discovered the default font that gets loaded has some corruption issue. and simply does not load. I used the opengl3 example as a base. After reverting to 1 commit earlier, the font showed up just fine (as opposed to black squares).

This is just a simple FYI, because it works fine for me with the commit earlier!

Thanks

commit that broke it: 98a38e24ab7036522a28964cd6983bd0c683742b

**Comment by Basez** [comment]
I must also note: the sample is not clearing the openGL state correctly after rendering, these should be added:

glDisable(GL_BLEND);
glEnable(GL_CULL_FACE);
glEnable(GL_DEPTH_TEST);

**Comment by ocornut** [comment]
How are you loading the texture ?
It works here in my test examples. I have optimized the texture agressively with pngout so it is probably a 1 or 2 bpp PNG file which certain loader (such as old version of stb_image.h) cannot read. I fixed stb_image.h recently to load low BPP textures.

**Comment by ocornut** [comment]
I'll probably eventually revert that shortly - because I am revamping the font system and the current change won't last anyway. But interested in figuring out if the issue is something else than PNG loading.

(PNG loading won't be needed with the upcoming font update)

**Comment by Basez** [comment]
Ah I think I know why this was causing issues in that case. I am also using SOIL in my engine which also uses the stb image loader, but probably an older version which you mentioned.

I discovered this because the #define STB_IMAGE_IMPLEMENTATION wouldn't work. (as it was already initialized by SOIL). I simply replaced the stbi_load_from_memory with SOIL_load_image_from_memory to fix that issue. The previous commit works fine for me and I already made some architectural changes to accommodate my needs, so I will probably continue to use this version! Thanks though :)
